### PR TITLE
Macos anka build farm doc 1727

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+ 
+ This enables connectivity between the Harness runner and the VMs. On the VM host, run the following command for each VM template:
+
+    anka modify $VM_NAME add port-forwarding service -g 9079
+
+For details, go to [Modifying your VM](https://docs.veertu.com/anka/anka-virtualization-cli/getting-started/modifying-your-vm/) in the Anka documentation.
+
+
+###  Set up the Anka controller and registry
+Set up the Anka registry and controller on a Linux host. For details, go to  [Setting up the Controller & Registry](https://docs.veertu.com/anka/anka-build-cloud/getting-started/setup-controller-and-registry/) in the Anka documentation.
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:50]: We should tell users what OS to run the controller/registry on. They can pick whatever they want - linux/Mac

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+###  Set up the Anka controller and registry
+Set up the Anka registry and controller on a Linux host. For details, go to  [Setting up the Controller & Registry](https://docs.veertu.com/anka/anka-build-cloud/getting-started/setup-controller-and-registry/) in the Anka documentation. 
+
+When you finish this workflow you will have:
+
+* An running instance of the Anka controller and registry.
+
+* A Mac node (such as a Mac Mini) in the Anka registry cluster. This node has Anka virtualization software installed. 
+
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:57]: remove registry - it's just anka cluster.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+* A VM template in the registry. You can use this to create VMs on other nodes as you add them to the registry. 
+
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:59]: as you add them to the cluster.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+:::note
+Harness recommends that you also enable token authentication on the registry host. For details, go to [Configuring Token Authentication](https://docs.veertu.com/anka/anka-build-cloud/advanced-security-features/token-authentication) in the Anka documentation.
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:61]: We shouldn't recommend - they can chose to have everything live on a private network therefore no need for auth. Something like - if you wish to enable auth for the controller & registry you can do it here...

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+::: 
+
+
+### Install the Harness delegate and runner
+Set up the Harness delegate and runner on the same node as your Anka controller and registry or on a separate node. 
+
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:67]: No - this isn't true. You don't have to run the delegate on the same machine. The delegate and runner can  live on the same machine. As long as the runner can connect the Anka Controller it's all good.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+You can run your delegate and runner on [Docker](define-a-docker-build-infrastructure.md), [MacOS](./define-macos-build-infra-with-anka-registry.md), [AWS](./set-up-an-aws-vm-build-infrastructure.md), [Azure](./define-a-ci-build-infrastructure-in-azure.md), and [Google Cloud Platform](./define-a-ci-build-infrastructure-in-google-cloud-platform.md) build infrastructures.
+
+
+### Update the Anka controller and Harness runner
+
+Do the following steps.
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:73]: Remove all this - this setup is described in the anka docs.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+#### Set up the Harness runner to communicate with the Anka controller
+
+On the Harness runner host, update up your pool.env file as follows:
+
+``` yaml
+  spec: 
+    account: 
+      username     : $ANKA_CONTROLLER_USERNAME
+      password     : $ANKA_CONTROLLER_PASSWORD_OR_TOKEN
+    controller_url : $ANKA_CONTROLLER_URL
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:113]: Incorrect yaml. please see my examples

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+```
+
+[Drone Runner config example](#harness-runner-poolyml-example)
+
+#### Restart the controller 
+
+Go back to the the docker-compose.yml folder and restart the controller: **`docker-compose -f docker-compose.yml up -d`**
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:120]: remove

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+Now that you've gone through the entire end-to-end workflow, you can add more nodes and templates as needed. After you join a node to the cluster, it can pull templates from the registry and use them to create nodes.
+
+:::note
+Before you push each VM to the registry, make sure that you enable port forwarding as described in [Set up port forwarding on the VM](#set-up-port-forwarding-on-the-vm) above.
+:::
+
+
+### YAML config examples
+
+#### Anka controller `docker-compose.yml` example
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:146]: remove

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+   platform:
+     os: darwin
+     arch: amd64
+   spec:
+     account:
+       username: anka-user
+       password: admin
+     vm_id: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxx 
+#    node_id: // create pool from specific node
+     controller_url: https://anka-controller.myorg.com:8089
```
Comment (eoinmcafee00) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:202]: we haven't updated this typo yet so it has to be registry_url but put a note it's the controller url you use.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+- [Install the Harness delegate and runner](#install-the-harness-delegate-and-runner)
+- [Set up the Harness runner to communicate with the Anka controller](#set-up-the-harness-runner-to-communicate-with-the-anka-controller)
+- [Set up the delegate in the Harness pipeline](#set-up-the-delegate-in-the-harness-pipeline)
+- [Add other Mac nodes and VM templates to the Anka registry](#add-other-mac-nodes-and-vm-templates-to-the-anka-registry)
+- [Harness runner `pool.yml` example](#harness-runner-poolyml-example)
+
+
+### Install Anka and create a VM on a Mac node
+ 
+ The first step is to set up the Anka software on one of your Mac nodes (such as a Mac Mini) and create a VM. For details, go to [Getting Started](https://docs.veertu.com/anka/anka-virtualization-cli/getting-started/) in the Anka documentation.
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:28]:
```suggestion
 The first step is to set up the Anka software on one of your Mac nodes (such as a Mac mini) and create a VM. For details, go to [Getting Started](https://docs.veertu.com/anka/anka-virtualization-cli/getting-started/) in the Anka documentation.
```

mini is lowercase in Mac mini (very consistent of them.... 🫠)

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+ When you finish this workflow, you will have a MacOS node with a working VM. After you set up the Anka registry, you can then push the node and the VM to the registry. 
+
+
+### Set up port forwarding on the VM
+ 
+ This enables connectivity between the Harness runner and the VMs. On the VM host, run the following command for each VM template:
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:35]:
```suggestion
 Port forwarding enables connectivity between the Harness runner and the VMs. On the VM host, run the following command for each VM template:
```

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+
+    anka modify $VM_NAME add port-forwarding service -g 9079
+
+For details, go to [Modifying your VM](https://docs.veertu.com/anka/anka-virtualization-cli/getting-started/modifying-your-vm/) in the Anka documentation.
+
+
+###  Set up the Anka controller and registry
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:42]:
```suggestion
###  Set up the Anka controller and registry

```

missing space after

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+Set up the Anka registry and controller. For details, go to  [Setting up the Controller & Registry](https://docs.veertu.com/anka/anka-build-cloud/getting-started/setup-controller-and-registry/) in the Anka documentation.
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:43]:
```suggestion
Set up the Anka registry and controller, as explained in the Anka documentation: [Setting up the Controller & Registry](https://docs.veertu.com/anka/anka-build-cloud/getting-started/setup-controller-and-registry/).
```

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+* A VM template in the registry. You can use this to create VMs on other nodes as you add them to the cluster. 
+
+:::note
+Optionally, you can enable token authentication for the controller and registry as described in [Configuring Token Authentication](https://docs.veertu.com/anka/anka-build-cloud/advanced-security-features/token-authentication) in the Anka documentation.
+::: 
+
+
+### Install the Harness delegate and runner
+Set up the Harness delegate and runner.
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:59]:
```suggestion
### Install the Harness delegate and runner

Set up the Harness delegate and runner.  
```

It is not immediately clear that the links go to instructions.

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+
+### Set up the Harness runner to communicate with the Anka controller
+
+On the Harness runner host, update up your pool.env file shown in the [Drone Runner config example](#harness-runner-poolyml-example) below.
+
+### Set up the delegate in the Harness pipeline
+
+In your Harness Project, go to a pipeline that includes a Build stage.
+
+In the Infrastructure tab of the Build Stage, define your infrastructure as follows:
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:72]:
```suggestion
In the **Infrastructure** tab of the **Build** stage, define your infrastructure as follows:
```

File: docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+
+###  Add other Mac nodes and VM templates to the Anka registry
+
+Now that you've gone through the entire end-to-end workflow, you can add more nodes and templates as needed. After you join a node to the cluster, it can pull templates from the registry and use them to create nodes.
+
+:::note
+Before you push each VM to the registry, make sure that you enable port forwarding as described in [Set up port forwarding on the VM](#set-up-port-forwarding-on-the-vm) above.
+:::
+
+### Harness runner `pool.yml` example
```
Comment (aimurphy) [inline docs/continuous-integration/use-ci/set-up-build-infrastructure/define-macos-build-infra-with-anka-registry.md:88]: Where do they use this in the workflow?

**Review by aimurphy — APPROVED** [review]
Some suggestions but approved. You do not have to take all my suggestions.

**Comment by douglas-j-bothwell** [comment]
Finalizing this doc in a new PR: https://github.com/harness/developer-hub/pull/732

**Pull request closed without merging**